modules/book.rb

```diff
@@ -4,11 +4,15 @@ class Book < Item
   attr_accessor :publisher, :cover_state
 
   def initialize(publisher, cover_state)
-    super()
+    super(publisher)
     @publisher = publisher
     @cover_state = cover_state
   end
 
+  def to_json(*_args)
+    { 'publisher' => @publisher, 'cover_state' => @cover_state }
```

Comment:
[OPTIONAL]
You have really worked hard on this, but it will be a best practice if you can use symbols in your hashes instead of the arrow (hash rocket).
You can do something like this
`{publisher: @publisher, cover_state: @cover_state}`